src/circuit/gadgets/case.rs

```diff
@@ -189,11 +187,94 @@ pub(crate) fn case<F: LurkField, CS: ConstraintSystem<F>>(
     multi_case(cs, selected, &[clauses], &[default], g).map(|r| r[0].to_owned())
 }
 
-/*
-Selects the clauses whose key is equal to selected and returns
-a vector containing the corresponding values, or if no key is selected,
-returns the default.
-*/
+///
+/// Selects the clauses whose key is equal to selected and returns
+/// a vector containing the corresponding values, or if no key is selected,
+/// returns the default value.
+///
+/// ```
+/// let mut cs = TestConstraintSystem::<Fr>::new();
+/// let s = &mut Store::<Fr>::default();
+/// let g = GlobalAllocations::new(&mut cs, s).unwrap();
+
+/// let x = Fr::from(123);
+/// let y = Fr::from(124);
+/// let z = Fr::from(125);
+/// let selected0 = AllocatedNum::alloc(cs.namespace(|| "selected0"), || Ok(x)).unwrap();
+/// let selected1 = AllocatedNum::alloc(cs.namespace(|| "selected1"), || Ok(z)).unwrap();
+/// let val0 = AllocatedNum::alloc(cs.namespace(|| "val0"), || Ok(Fr::from(666))).unwrap();
+/// let val1 = AllocatedNum::alloc(cs.namespace(|| "val1"), || Ok(Fr::from(777))).unwrap();
+/// let val2 = AllocatedNum::alloc(cs.namespace(|| "val2"), || Ok(Fr::from(700))).unwrap();
+/// let default_vec = [
+///     &AllocatedNum::alloc(cs.namespace(|| "default0"), || Ok(Fr::from(999))).unwrap(),
+///     &AllocatedNum::alloc(cs.namespace(|| "default1"), || Ok(Fr::from(998))).unwrap(),
+///     &AllocatedNum::alloc(cs.namespace(|| "default2"), || Ok(Fr::from(997))).unwrap(),
```

Comment:
You'd typically try to make this shorter by importing `AllocatedNum::alloc`

Reply:
Maybe in such a test, the import makes sense, but in general (in the code itself) we wouldn't import `alloc`, since many types implement such a method, and we usually want to know what type is being allocated.